test/expect-tests/fiber/fiber_tests.ml

```diff
@@ -655,3 +655,35 @@ let%expect_test "Sequence.parallel_iter doesn't leak" =
   test ~iter_function:Fiber.Sequence.parallel_iter ~check:(fun ~prev ~curr ->
       prev = curr);
   [%expect {| PASS |}]
+
+let%expect_test "fork - exceptions always thrown" =
+  test (backtrace_result unit)
+    (Fiber.collect_errors (fun () ->
+         Fiber.fork_and_join_unit
+           (fun () -> failwith "left")
+           (fun () -> failwith "right")));
```

Comment:
I assume this test is deterministic. However, looking at the docs in `fiber.mli`, I see no written guarantees. Could you document this behaviour?

Reply:
Well, my test just cares that _all_ exceptions are caught -  in whatever order. The expect test itself obviously relies on determinism for the assertion to work, but I'm not sure it's a property we want to guarantee. I could also just sort the list of exceptions, but it seems a bit unnecessary.

Reply:
> Well, my test just cares that all exceptions are caught - in whatever order.

That's not true! Right now your test very much cares about the order: it explicitly dictates it :-) If you really mean to write a test that doesn't care about the order, then sorting the results seems to be the right thing to do.

Reply:
I sorted the tests.